os/arch/arm/src/armv7-a/Toolchain.defs

```diff
@@ -0,0 +1,239 @@
+#  either express or implied. See the License for the specific
+#  language governing permissions and limitations under the License.
+# 
+############################################################################
+
+############################################################################
+# arch/arm/src/armv7-a/Toolchain.defs
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+# Setup for the selected toolchain
+
+#
+# Select and allow the selected toolchain to be overridden by a command-line
+#selection.
+#
+
+ifeq ($(filter y, \
+      $(CONFIG_ARMV7A_TOOLCHAIN_BUILDROOT) \
+    ),y)
+  CONFIG_ARMV7A_TOOLCHAIN ?= BUILDROOT
+endif
+
+ifeq ($(filter y, \
+      $(CONFIG_ARMV7A_TOOLCHAIN_GNU_EABI) \
+    ),y)
+  CONFIG_ARMV7A_TOOLCHAIN ?= GNU_EABI
+endif
+
+#
+# Supported toolchains
+#
+# Each toolchain definition should set:
+#
+#  CROSSDEV         The GNU toolchain triple (command prefix)
+#  ARCHCPUFLAGS     CPU-specific flags selecting the instruction set
+#                   FPU options, etc.
+#  ARCHOPTIMIZATION The optimization level that results in
+#                   reliable code generation.
+#
+
+ifeq ($(CONFIG_ARM_THUMB),y)
+  ARCHCPUFLAGS += -mthumb
+endif
+
+ifeq ($(CONFIG_ARCH_CORTEXA5),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a5
+else ifeq ($(CONFIG_ARCH_CORTEXA7),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a7
+else ifeq ($(CONFIG_ARCH_CORTEXA8),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a8
+else ifeq ($(CONFIG_ARCH_CORTEXA9),y)
+  ARCHCPUFLAGS += -mcpu=cortex-a9
+endif
+
+ifneq ($(CONFIG_ARM_DPFPU32),y)
+  ARCHFPUD16 = -d16
+endif
+
+# Cortex-A5  | -mfpu=vfpv4-fp16 | -mfpu=vfpv4-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A7  | -mfpu=vfpv4      | -mfpu=vfpv4-d16      | -mfpu=neon-vfpv4
+# Cortex-A8  | -mfpu=vfpv3      |                      | -mfpu=neon (alias for neon-vfpv3)
+# Cortex-A9  | -mfpu=vfpv3-fp16 | -mfpu=vfpv3-d16-fp16 | -mfpu=neon-fp16
+# Cortex-A15 | -mfpu=vfpv4      |                      | -mfpu=neon-vfpv4
+
+ifeq ($(CONFIG_ARCH_FPU),y)
+  ifeq ($(CONFIG_ARM_FPU_ABI_SOFT),y)
+    ARCHCPUFLAGS += -mfloat-abi=softfp
+  else
+    ARCHCPUFLAGS += -mfloat-abi=hard
+  endif
+
```

Comment:
There was a flag for neon here. We removed it because it was causing error on some compilers. Can you please check if there is a way to see the current compiler version and add the neon flag if the compiler supports it?